wandb/filesync/dir_watcher.py

```diff
@@ -110,24 +108,36 @@ def __init__(
         file_path: PathStr,
         save_name: SaveName,
         file_pusher: FilePusher,
+        settings: SettingsStatic,
         *args,
-        **kwargs
+        **kwargs,
     ):
         super().__init__(file_path, save_name, file_pusher, *args, **kwargs)
         self._last_uploaded_time = None
         self._last_uploaded_size = 0
+        try:
+            self.RATE_LIMIT_SECONDS = (
+                settings._live_policy_rate_limit or self.RATE_LIMIT_SECONDS
+            )
+        except AttributeError:
```

Comment:
When would you hit this AttributeError? When the base Settings object (for the SettingsStatic object) didn't have that? Could we make it non-optional, but always default to None instead in the Static case?

Reply:
I don't understand your suggestion ... what do you mean?

Reply:
I was just trying to figure out when `settings._live_policy_rate_limit` wouldn't exist and result in an attribute error. As in, when would SettingsStatic not have this attribute?

Reply:
oh I see... so yeah it seems like when we sync old examples like the one I linked above it happens

Reply:
i dont understand that case..  we should not be pulling settings from a old sync file, if we are, we are going to run into some problems with all settings changes.

Reply:
I was running the test and it was failing on the missing attributes... I can try and look further into it

Reply:
ok found the issue... in sender we needed to explicitly fix the setting since it uses dict and not settings, left a TODO to fix this.